apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py

```diff
@@ -760,6 +788,26 @@ def grade(element_html: str, data: pl.QuestionData) -> None:
             None if num_initial_correct == len(submission) else num_initial_correct
         )
 
+        if feedback_type in FIRST_WRONG_TYPES:
+            for block in student_answer[:num_initial_correct]:
+                block["badge_type"] = "badge-success"
+                block["icon"] = "fa-check"
+                block["distractor_feedback"] = ""
+
+            if first_wrong is not None:
+                student_answer[first_wrong]["badge_type"] = "badge-danger"
+                student_answer[first_wrong]["icon"] = "fa-xmark"
+                if feedback_type is not FeedbackType.FIRST_WRONG_VERBOSE:
+                    student_answer[first_wrong]["distractor_feedback"] = ""
```

Comment:
Is this necessary? Seems strange to delete the data given in the student answer rather than just not display it later on somewhere.

Reply:
Whether or not it is displayed depends on the value of `first_wrong`, which (now that we removed it from the `partial_scores` dict), is only accessible here in the `grade` function and not later on in `render`. That's why I figured this might be the cleanest way to do it.